File: backend/masz/Models/AutoModerationConfig.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Runtime.Serialization;
using masz.Dtos.ModCase;

namespace masz.Models
{
    public class AutoModerationConfig
    {
        [Key]
        public int Id { get; set; }
        public string GuildId { get; set; }
        public AutoModerationType AutoModerationType { get; set; }
        public AutoModerationAction AutoModerationAction { get; set; }
        public PunishmentType? PunishmentType { get; set; }
        public int? PunishmentDurationMinutes { get; set; }
        public string[] IgnoreChannels { get; set; }
        public string[] IgnoreRoles { get; set; }
        public int? TimeLimitMinutes { get; set; }
        public int? Limit { get; set; }
        public string CustomWordFilter { get; set; }
        public bool SendDmNotification { get; set; }
        public bool SendPublicNotification { get; set; }
    }
}

File: nginx/MASZ/src/app/models/AutoModerationConfig.ts
import { AutoModerationAction } from "./AutoModerationAction";
import { AutoModerationType } from "./AutoModerationType";
import { ChannelNotificationBehavior } from "./ChannelNotificationBehavior";
import { PunishmentType } from "./PunishmentType";

export interface AutoModerationConfig {
    id: number;
    guildId: string;
    autoModerationType: AutoModerationType;
    autoModerationAction: AutoModerationAction;
    punishmentType: PunishmentType;
    punishmentDurationMinutes: number;
    ignoreChannels: string[];
    ignoreRoles: string[];
    timeLimitMinutes?: number;
    limit?: number;
    customWordFilter?: string;
    sendDmNotification: boolean;
    sendPublicNotification: boolean;
    channelNotificationBehavior: ChannelNotificationBehavior;
}

